packages/components/src/internal/components/permissions/PermissionAssignments.tsx

```diff
@@ -1,218 +1,275 @@
-                                            {this.renderSaveButton()}
-                                        </Alert>
-                                    </div>
-                                )
-                            ) : (
-                                <div className="permissions-save-alert">
-                                    <Alert bsStyle="info">
-                                        Permissions for this container are being inherited from its parent.
-                                    </Alert>
-                                </div>
-                            )}
-                            {visibleRoles.map((role, i) => {
-                                return (
-                                    <PermissionsRole
-                                        key={i}
-                                        role={role}
-                                        assignments={policy.assignmentsByRole.get(role.uniqueName)}
-                                        typeToShow={typeToShow}
-                                        principals={principals}
-                                        onClickAssignment={this.showDetails}
-                                        onRemoveAssignment={isEditable ? this.removeAssignment : undefined}
-                                        onAddAssignment={isEditable ? this.addAssignment : undefined}
-                                        selectedUserId={selectedUserId}
-                                        disabledId={disabledId}
-                                    />
-                                );
-                            })}
-                            <br />
-                            {saveErrorMsg && <Alert>{saveErrorMsg}</Alert>}
-                            {isEditable && this.renderSaveButton()}
-                        </Panel.Body>
-                    </Panel>
-                </Col>
-                {showDetailsPanel && (
-                    <Col xs={12} md={4}>
-                        {selectedPrincipal && selectedPrincipal.type === 'g' ? (
-                            <GroupDetailsPanel
-                                principal={selectedPrincipal}
-                                policy={policy}
-                                rolesByUniqueName={rolesByUniqueName}
-                            />
-                        ) : (
-                            <UserDetailsPanel
-                                userId={selectedUserId}
-                                policy={policy}
-                                rootPolicy={rootPolicy}
-                                rolesByUniqueName={rolesByUniqueName}
-                            />
-                        )}
-                    </Col>
-                )}
-            </Row>
-        );
+        setSubmitting(true);
+
+        // Policy has been switched to inherited. Delete the current policy.
+        if (inherited && !wasInherited) {
+            Security.deletePolicy({
```

Comment:
Should we consider adding these methods (deletePolicy, savePolicy) to the Security API wrapper, which would allow us to expose them as async functions and use `await`?